transport-native-epoll/src/main/java/io/netty/channel/epoll/AbstractEpollChannel.java

```diff
@@ -492,5 +545,232 @@ protected final void clearEpollIn0() {
                 unsafe().close(unsafe().voidPromise());
             }
         }
+
+        @Override
+        public void connect(
+                final SocketAddress remoteAddress, final SocketAddress localAddress, final ChannelPromise promise) {
+            if (!promise.setUncancellable() || !ensureOpen(promise)) {
```

Comment:
`ensureOpen` is deprecated [1]. I'm not sure why but can we use a non-deprecated method?

[1] https://github.com/netty/netty/commit/e84567004324c0bfd04b0ca9e1e182ed89394b16#diff-cfab6ab9111be41b2f39beb8d154e79fR874